Description: Use shared libraries available in Debian.
Author: Andriy Senkovych <jolly_roger@itblog.org.ua>
Last-Update: 2013-08-28
--- a/Makefile
+++ b/Makefile
@@ -1,11 +1,10 @@
 OUT=webdis
-HIREDIS_OBJ=hiredis/hiredis.o hiredis/sds.o hiredis/net.o hiredis/async.o
-JANSSON_OBJ=jansson/src/dump.o jansson/src/error.o jansson/src/hashtable.o jansson/src/load.o jansson/src/strbuffer.o jansson/src/utf.o jansson/src/value.o jansson/src/variadic.o
 FORMAT_OBJS=formats/json.o formats/raw.o formats/common.o formats/custom-type.o
 HTTP_PARSER_OBJS=http-parser/http_parser.o
 
-CFLAGS=-O0 -ggdb -Wall -Wextra -I. -Ijansson/src -Ihttp-parser
-LDFLAGS=-levent -pthread
+CFLAGS ?= -O0 -ggdb
+CFLAGS += -Wall -Wextra -I. -Ihttp-parser -Ib64
+LDFLAGS += -levent -pthread -lhiredis -ljansson -lb64
 
 # check for MessagePack
 MSGPACK_LIB=$(shell ls /usr/lib/libmsgpack.so 2>/dev/null)
@@ -16,28 +15,28 @@
 endif
 
 
-DEPS=$(FORMAT_OBJS) $(HIREDIS_OBJ) $(JANSSON_OBJ) $(HTTP_PARSER_OBJS)
-OBJS=webdis.o cmd.o worker.o slog.o server.o libb64/cencode.o acl.o md5/md5.o sha1/sha1.o http.o client.o websocket.o pool.o conf.o $(DEPS)
+DEPS=$(FORMAT_OBJS) $(HTTP_PARSER_OBJS)
+OBJS=webdis.o cmd.o worker.o slog.o server.o acl.o md5/md5.o sha1/sha1.o http.o client.o websocket.o pool.o conf.o $(DEPS)
 
 
 
 PREFIX ?= /usr/local
-CONFDIR ?= $(DESTDIR)/etc
+CONFDIR ?= $(DESTDIR)/etc/webdis
 
 INSTALL_DIRS = $(DESTDIR) \
 	       $(DESTDIR)/$(PREFIX) \
 	       $(DESTDIR)/$(PREFIX)/bin \
 	       $(CONFDIR)
 
-all: $(OUT) Makefile
+all: $(OUT)
 
-$(OUT): $(OBJS) Makefile
+$(OUT): $(OBJS)
 	$(CC) -o $(OUT) $(OBJS) $(LDFLAGS)
 
-%.o: %.c %.h Makefile
+%.o: %.c %.h
 	$(CC) -c $(CFLAGS) -o $@ $<
 
-%.o: %.c Makefile
+%.o: %.c
 	$(CC) -c $(CFLAGS) -o $@ $<
 
 $(INSTALL_DIRS):
@@ -45,7 +44,8 @@
 
 clean:
 	rm -f $(OBJS) $(OUT)
+	cd tests && $(MAKE) $@
 
 install: $(OUT) $(INSTALL_DIRS)
 	cp $(OUT) $(DESTDIR)/$(PREFIX)/bin
-	cp webdis.prod.json $(CONFDIR)
+	cp webdis.prod.json $(CONFDIR)/webdis.json
--- a/conf.c
+++ b/conf.c
@@ -9,7 +9,7 @@
 
 #include <jansson.h>
 #include <evhttp.h>
-#include <libb64/cencode.h>
+#include <b64/cencode.h>
 #include "conf.h"
 #include "acl.h"
 
--- a/websocket.c
+++ b/websocket.c
@@ -1,5 +1,5 @@
 #include "sha1/sha1.h"
-#include "libb64/cencode.h"
+#include <b64/cencode.h>
 #include "websocket.h"
 #include "client.h"
 #include "cmd.h"
